python/osrd_schemas/scripts/truncate_infra.py

```diff
@@ -0,0 +1,378 @@
+                update={"track_ranges": new_track_ranges}
+            )
+            res.append(new_speed_section)
+
+    return res
+
+
+def filter_neutral_sections(
+    neutral_sections: list[NeutralSection], track_section_ids: set[str]
+) -> list[NeutralSection]:
+    res = []
+    for neutral_section in neutral_sections:
+        new_track_ranges = filter_track_ranges(
+            neutral_section.track_ranges, track_section_ids
+        )
+        if new_track_ranges:
+            new_announcement_ranges = filter_track_ranges(
+                neutral_section.announcement_track_ranges, track_section_ids
+            )
+            new_neutral_section = neutral_section.model_copy(
+                update={
+                    "track_ranges": new_track_ranges,
+                    "announcement_track_ranges": new_announcement_ranges,
+                }
+            )
+            res.append(new_neutral_section)
+
+    return res
+
+
+def filter_switches(
+    switches: list[Switch],
+    track_section_ids: set[str],
+    track_length_by_track: dict[str, float],
+) -> tuple[list[Switch], list[BufferStop], dict[str, tuple[str, Direction]]]:
+    """Remove switches that are not on the track sections to keep and add buffer stops if needed.
+
+    Returns the new list of switches, the list of buffer stops created and a dict mapping the ids of former switches to
+    the ids of the buffer stops that replaced them and the direction the buffer stops are facing.
+    """
+
+    filtered_switches = []
+    new_buffer_stops = []
+    former_switch_id_to_buffer_stop_id = {}
+    for switch in switches:
+        n_ports_included = sum(
+            port.track in track_section_ids for port in switch.ports.values()
+        )
+        if n_ports_included == len(switch.ports):
+            # All ports are included, no need to change anything
+            filtered_switches.append(switch)
+        elif n_ports_included > 1:
+            print(
+                f"Switch {switch.id} should be included in the bbox, but isn't handled yet"
+            )
+        elif n_ports_included == 1:
+            # Only one port is included, we need to replace the switch by a buffer stop
+            the_port = next(
+                port
+                for port in switch.ports.values()
+                if port.track in track_section_ids
+            )
+            new_buffer_stops.append(
+                BufferStop(
+                    id=f"buffer_stop_{switch.id}",
+                    track=the_port.track,
+                    position=0
+                    if the_port.endpoint == Endpoint.BEGIN
+                    else track_length_by_track[the_port.track],
+                )
+            )
+            former_switch_id_to_buffer_stop_id[switch.id] = (
+                new_buffer_stops[-1].id,
+                Direction.START_TO_STOP
+                if the_port.endpoint == Endpoint.BEGIN
+                else Direction.STOP_TO_START,
+            )
```

Comment:
nitpick: To avoid indexing in the list something already appended, you could do this


```suggestion
            new_buffer_stop = BufferStop(
                id=f"buffer_stop_{switch.id}",
                track=the_port.track,
                position=0
                if the_port.endpoint == Endpoint.BEGIN
                else track_length_by_track[the_port.track],
            )
            former_switch_id_to_buffer_stop_id[switch.id] = (
                new_buffer_stop.id,
                Direction.START_TO_STOP
                if the_port.endpoint == Endpoint.BEGIN
                else Direction.STOP_TO_START,
            )
            new_buffer_stops.append(new_buffer_stop)
```

Reply:
Done in https://github.com/OpenRailAssociation/osrd/pull/13526/commits/bddc5efbd130f7e85dc7d9d5f400270b7febf6e0.